101-datastore-create-blob/azuredeploy.json

```diff
@@ -0,0 +1,110 @@
+{
+	"$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+	"contentVersion": "1.0.0.0",
+	"parameters": {
+		"workspaceName": {
+			"type": "string",
+			"metadata": {
+				"description": "Specifies the name of the Azure Machine Learning workspace which will hold this datastore target."
+			}
+		},
+		"datastoreName": {
+			"type": "string",
+			"metadata": {
+				"description": "The name of the datastore, case insensitive, can only contain alphanumeric characters and underscore"
+			}
+		},
+		"accountName": {
+			"type": "string",
+			"metadata": {
+				"description": "The name of the storage account."
+			}
+		},
+		"containerName": {
+			"type": "string",
+			"metadata": {
+				"description": "The name of the azure blob container."
+			}
+		},
+		"storageAccountSubscriptionId": {
+			"type": "string",
+			"defaultValue":"",
+			"metadata": {
+				"description": "Optional : The subscription ID of the storage account."
+			}
+		},
+		"storageAccountResourceGroup": {
+			"type": "string",
+			"defaultValue":"",
+			"metadata": {
+				"description": "Optional : The resource group of the storage account."
+			}
+		},
+		"accountKey": {
+			"type": "securestring",
+			"defaultValue":"",
+			"metadata": {
+				"description": "Optional : A storage account key, defaults to None. Skip this if SAS token authentication is used."
+			}
+		},
+		"sasToken": {
+			"type": "securestring",
+			"defaultValue":"",
+			"metadata": {
+				"description": "Optional : An account SAS token, defaults to None. Skip this if storage account key authentication is used."
+			}
+		},
+		"endPoint": {
+			"type": "string",
+			"defaultValue":"",
+			"metadata": {
+				"description": "Optional : The endpoint of the storage account. Defaults to core.windows.net"
```

Comment:
does the user have to provide this?  we could get it using reference or the environment() function...

Reply:
This is optional for customers, we set defaults on our service side to avoid making changes on multiple clients. (UI, CLI, SDK)